resources/scripts/components/server/Console.tsx

```diff
@@ -82,26 +88,34 @@ export default () => {
     useEffect(() => {
         if (connected && ref.current && !terminal.element) {
             terminal.open(ref.current);
+            terminal.loadAddon(fitAddon);
+            terminal.loadAddon(searchAddon);
+            terminal.loadAddon(searchAddonBar);
+            fitAddon.fit();
 
-            // @see https://github.com/xtermjs/xterm.js/issues/2265
-            // @see https://github.com/xtermjs/xterm.js/issues/2230
-            TerminalFit.fit(terminal);
-
-            // Add support for copying terminal text.
+            // Add support for capturing keys
             terminal.attachCustomKeyEventHandler((e: KeyboardEvent) => {
-                // Ctrl + C
+            // Ctrl + C ( Copy )
                 if (e.ctrlKey && (e.key === 'c')) {
                     document.execCommand('copy');
                     return false;
                 }
 
+                if (e.ctrlKey && (e.key === 'f')) {
+                    searchAddonBar.show();
+                    return false;
+                }
+
+                if (e.key === 'Escape') {
+                    searchAddonBar.hidden();
+                }
                 return true;
             });
         }
     }, [ terminal, connected ]);
 
     const fit = debounce(() => {
-        TerminalFit.fit(terminal);
+        fitAddon.fit();
```

Comment:
If we have this, do we need the previous call to `fit()`? Or does it lead to the console taking 100ms to adapt if the first one isn't there?

Reply:
I have no idea? it was already like that I just adapted it to the new way Xterm does addons in 4.x

Reply:
When initially rendering xterm you should call `fit()` to ensure that the terminal is properly sized on component load, otherwise you can end up with weird results when the resize event is called.  The code for the fitAddon seems fine to me.

Reply:
This was added by Dane to resize the terminal when the panel was being made responsive.

Reply:
Nevermind, I misunderstood that piece of code, sorry 😄